client/handler.go

```diff
@@ -122,6 +134,70 @@ func (h *Handler) Create(w http.ResponseWriter, r *http.Request, _ httprouter.Pa
 	h.r.Writer().WriteCreated(w, r, ClientsHandlerPath+"/"+c.GetID(), &c)
 }
 
+// swagger:route POST /dyn-clients public createDynOAuth2Client
+//
+// Create an OAuth 2.0 Client
+//
+// Create a new OAuth 2.0 client If you pass `client_secret` the secret will be used, otherwise a random secret will be generated. The secret will be returned in the response and you will not be able to retrieve it later on. Write the secret down and keep it somwhere safe.
+//
+// OAuth 2.0 clients are used to perform OAuth 2.0 and OpenID Connect flows. Usually, OAuth 2.0 clients are generated for applications which want to consume your OAuth 2.0 or OpenID Connect capabilities. To manage ORY Hydra, you will need an OAuth 2.0 Client as well. Make sure that this endpoint is well protected and only callable by first-party components.
+//
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       201: oAuth2Client
+//       400: genericError
+//       409: genericError
+//       500: genericError
+func (h *Handler) CreateDynamicRegistration(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var c Client
+
+	if err := json.NewDecoder(r.Body).Decode(&c); err != nil {
```

Comment:
Looks like a lot of code is copy & paste. Could you please reuse the existing logic by extracting it into common functions or something similar?